### PR TITLE
Use an empty vendored file system in Ruff

### PR DESCRIPTION
**Pull request by MichaReiser**
## Summary

This PR changes removes the typeshed stubs from the vendored file system shipped with ruff
and instead ships an empty "typeshed".

Making the typeshed files optional required extracting the typshed files into a new `ruff_vendored` crate. I do like this even if all our builds always include typeshed because it means `red_knot_python_semantic` contains less code that needs compiling.

This also allows us to use deflate because the compression algorithm doesn't matter for an archive containing a single, empty file. 

## Test Plan

`cargo test`

I verified with ` cargo tree -f "{p} {f}" -p <package> ` that:

* red_knot_wasm: enables `deflate` compression
* red_knot: enables `zstd` compression
* `ruff`: uses stored

I'm not quiet sure how to build the binary that maturin builds but comparing the release artifact size with `strip = true` shows a `1.5MB` size reduction

### CODE REVIEW

**Comment by MichaReiser** [comment]
@charliermarsh this is a potential short term fix for the conda forge build

**Review by charliermarsh — APPROVED** [review]

**Comment by charliermarsh** [comment]
This appears to be working: https://github.com/conda-forge/ruff-feedstock/pull/220

**Comment by charliermarsh** [comment]
I'm happy to release it today if you want

**Comment by MichaReiser** [comment]
Nice. Could you try running the new analyze method over a real project? Just to make sure we don't show annoying warnings when the versions file is empty :laughing:

**Comment by MichaReiser** [comment]
But yeah. Feel free to merge and release the changes to unblock cargo forge.

**Comment by MichaReiser** [comment]
For changelog: Fix conda forge build.

**Comment by MichaReiser** [comment]
> This appears to be working: [conda-forge/ruff-feedstock#220](https://github.com/conda-forge/ruff-feedstock/pull/220)

I love your confidence :laughing:

File: crates/ruff_db/src/.vendored.rs.pending-snap
Comment (AlexWaygood) [inline crates/ruff_db/src/.vendored.rs.pending-snap:1]: This file shouldn't be here

**Comment by charliermarsh** [comment]
Just ran, no warnings :)

File: crates/ruff_db/src/.vendored.rs.pending-snap
Comment (MichaReiser) [reply crates/ruff_db/src/.vendored.rs.pending-snap:1]: Thanks. I don't understand why it sometimes creates those files but doesn't delete them when the test is done.

**Pull request merged**